File: app/src/main/java/com/samet/coinApp/network/APIService.java
package com.samet.coinApp.network;


import com.samet.coinApp.models.CryptoList;
import com.samet.coinApp.models.Info;

import io.reactivex.Observable;
import retrofit2.http.GET;
import retrofit2.http.Query;

public interface APIService {

    @GET("/v1/cryptocurrency/listings/latest?")
    Observable<CryptoList> getMarketPairsLatest(@Query("start") String start, @Query("limit") String limit);

    @GET("/v1/cryptocurrency/info")
    Observable<Info> getCryptocurrencyInfo(@Query("symbol") String symbol);
}

File: app/src/main/java/com/samet/coinApp/api/ApiService.kt
package com.samet.coinApp.api

import com.samet.coinApp.models.CryptoList
import com.samet.coinApp.models.Info
import retrofit2.http.GET
import retrofit2.http.Query

interface ApiService {
    @GET("/v1/cryptocurrency/listings/latest?")
    suspend fun getCoinListLatest(
        @Query("start") start: String?,
        @Query("limit") limit: String?
    ): CryptoList?

    @GET("/v1/cryptocurrency/info")
    suspend fun getCryptoCurrencyInfo(@Query("symbol") symbol: String?): Info?
}